### PR TITLE
Ng 11 view register values

### PR DESCRIPTION
**Pull request by cfboy**

### CODE REVIEW

**Review by alejoreyes96 — APPROVED** [review]

**Review by noelvalentin — APPROVED** [review]

File: GUI/window.py [hunk 1]
```diff
@@ -67,6 +61,7 @@ def __init__(self, **kwargs):
         self.app = kwargs.pop('app')
         self.micro_sim = kwargs.pop('micro_sim')
         self.step_index = 0
+        self.register = kwargs.pop('registers')
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove this line

File: GUI/window.py [hunk 2]
```diff
@@ -120,14 +119,17 @@ def run_micro_instructions_step(self, instance):
...
+        self.register_table.get_data()
+
+
 class MainWindow(BoxLayout):
 
     def __init__(self, **kwargs):
         self.nav_drawer = kwargs.pop('nav_drawer')
         self.app = kwargs.pop('app')
         self.micro_sim = kwargs.pop('micro_sim')
+        self.register = kwargs.pop('registers')
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove this line

File: GUI/window.py [hunk 3]
```diff
@@ -140,7 +142,7 @@ def __init__(self, **kwargs):
                                   left_action_items=[['dots-vertical', lambda x: self.nav_drawer.toggle_nav_drawer()]]))
 
         self.add_widget(BoxLayout())  # Bumps up navigation bar to the top
-        self.add_widget(RunWindow(app=self.app, micro_sim=self.micro_sim))
+        self.add_widget(RunWindow(app=self.app, micro_sim=self.micro_sim, registers=self.register))
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove ‘registers’

File: GUI/window.py [hunk 4]
```diff
@@ -264,7 +245,7 @@ def __init__(self, **kwargs):
         self.app = App.get_running_app()
         self.micro_sim = MicroSim()
         self.add_widget(NavDrawer(micro_sim=self.micro_sim))
-        self.add_widget(MainWindow(nav_drawer=self, app=self.app, micro_sim=self.micro_sim))
+        self.add_widget(MainWindow(nav_drawer=self, app=self.app, micro_sim=self.micro_sim, registers=REGISTER))
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove ‘registers’

File: microprocessor_simulator.py [hunk 5]
```diff
@@ -80,34 +78,29 @@ def run_micro_instructions_step(self, step_index):
         f.write("\n\n First 50 slots in memory: \n\n")
         i = 0
         for m in range(50):
-                print(f'{RAM[i]} {RAM[i + 1]}')
-                f.write(f'{RAM[i]} {RAM[i + 1]}' + '\n')
+            print(f'{RAM[i]} {RAM[i + 1]}')
```
Comment (ChristianPerez34) [inline microprocessor_simulator.py]: Remove print

File: microprocessor_simulator.py [hunk 6]
```diff
@@ -80,34 +78,29 @@ def run_micro_instructions_step(self, step_index):
...
         self.index = 0
         self.is_running = True
         self.cond = False
-        self.stack_pointer = 0
-        self.program_counter = 0
-        self.false_index = -1
-
-
-
+        REGISTER['sp'] = f'{0:02x}'
```
Comment (ChristianPerez34) [inline microprocessor_simulator.py]: Should clear all registers. General purpose registers, program counter, etc

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: GUI/window.py (hunk 1 above)
Comment (cfboy) [reply GUI/window.py]: Ok

File: GUI/window.py [hunk 7]
```diff
@@ -198,64 +198,43 @@ def events(self, instance, keyboard, keycode, text, modifiers):
...
-            self.data.append(f'{RAM[i]} {RAM[i + 1]}')
-            i += 2
-        
-        table.get_data(self.data)
-        
-        # print(self.micro_sim.micro_instructions)
 
 
-class Table1(RecycleView):
+class register_table(RecycleView):
```
Comment (ChristianPerez34) [inline GUI/window.py]: `class RegisterTable(RecycleView)`

File: GUI/window.py [hunk 8]
```diff
@@ -198,64 +198,43 @@ def events(self, instance, keyboard, keycode, text, modifiers):
...
     data_list = ListProperty([])
 
-
     def __init__(self, **kwargs):
-        super(Table1, self).__init__(**kwargs)
+        # self.register = REGISTER
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove comment

File: microprocessor_simulator.py [hunk 9]
```diff
@@ -80,34 +78,30 @@ def run_micro_instructions_step(self, step_index):
...
         self.index = 0
         self.is_running = True
         self.cond = False
-        self.stack_pointer = 0
-        self.program_counter = 0
-        self.false_index = -1
-
-
-
+        REGISTER['pc'] = f'{0:03x}'
```
Comment (ChristianPerez34) [inline microprocessor_simulator.py]:
```Python
for k, v in REGISTER.items():
    if k == 'pc' or k == 'sp':
        REGISTER[k] = f'{0:03x}'
    elif k == 'ir':
        REGISTER[k] =  f'{0:04x}'
    else:
        REGISTER[k] = f'{0:02x}'
```

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]
Minor changes needed

**Review by ChristianPerez34 — APPROVED** [review]

**Pull request merged**